### PR TITLE
Subscribe to repo on mount

### PR DESCRIPTION
**Pull request by nhunzaker**
https://twitter.com/dan_abramov/status/790581793397305345:

> Don’t subscribe to data sources in constructor or componentWillMount(). Use componentDidMount(), teardown in componentWillUnmount().

So we don't, anymore. I've also removed `this.pure` in favor of `this.isImpure()`.

### CODE REVIEW

File: src/addons/presenter.js [hunk 1]
```diff
@@ -21,8 +19,8 @@ function wrappedRender () {
  */
 class Presenter extends React.Component {
 
-  constructor() {
-    super(...arguments)
+  constructor(props, context) {
+    super(props, context)
```
Comment (greypants) [inline src/addons/presenter.js]: 👍 for clarity

**Comment by greypants** [comment]
That's a fun twitter thread.

**Comment by ltk** [comment]
:+1:

**Comment by nhunzaker** [comment]
For whatever it's worth, each Presenter gets a fork of the parent repo. Subscriptions are localized to that repo. I'm not actually sure we _can_ leak event subscriptions.

**Comment by nhunzaker** [comment]
Thanks everyone!

**Pull request merged**